wgsl/index.bs

```diff
@@ -4558,20 +4558,26 @@ WGSL defines two types of expressions that can be evaluated before runtime:
 
 ### `const` Expressions ### {#const-expr}
 
-Expressions that are evaluated at [=shader module creation|shader-creation
+Expressions that can be evaluated at [=shader module creation|shader-creation
 time=] are called <dfn noexport>const-expressions</dfn>.
-In order for an expression to be evaluated at shader-creation time all
-[=identifiers=] in the expression must [=resolve=] to:
+An expression is a const-expression if all its [=identifiers=] [=resolve=] to:
 * [=const-declarations=], or
 * [=const-functions=], or
 * [=type aliases=], or
-* [=structure=] names
+* [=structure=] names.
 
-The type of a `const` expression must [=type rules|resolve=] to a type with a
+The type of a `const` expression [=shader-creation error|must=] [=type rules|resolve=] to a type with a
 [=creation-fixed footprint=].
 
 Note: [=type/abstract|Abstract types=] can be the inferred type of a const-expression.
 
+A const-expression |E| will be evaluated if and only if:
+* |E| is [=top-level expression=], or
+* |E| is a [=subexpression=] of a [=top-level expression=] |TE|, and evaluation of |TE| requires |E| to be evaluated.
```

Comment:
Maybe this doesn't need top-level expression. Otherwise it feels like you've solved only level 1 instead of all levels.
```suggestion
* |E| is a [=subexpression=] of an expression being evaluated |PE|, and evaluation of |PE| requires |E| to be evaluated.
```

Reply:
Also added a clause that the parent expression "will" be evaluated.